pennylane/devices/default_clifford.py

```diff
@@ -460,57 +519,239 @@ def simulate(
                         flat_state = qml.math.flatten(state)
                         debugger.snapshots[op.tag or len(debugger.snapshots)] = flat_state
 
+        stim_circuit = initial_circuit + stim_ct
         tableau_simulator.do_circuit(stim_ct)
-
         global_phase = qml.GlobalPhase(qml.math.sum(op.data[0] for op in global_phase_ops))
-        return self.measure(circuit, tableau_simulator, global_phase, stim)
+
+        # Perform measurments based on whether shots are provided
+        if circuit.shots:
+            meas_results = self.measure_statistical(circuit, stim_circuit, stim, seed=seed)
+        else:
+            meas_results = self.measure_analytical(
+                circuit, stim_circuit, tableau_simulator, global_phase, stim
+            )
+
+        return meas_results[0] if len(meas_results) == 1 else tuple(meas_results)
 
     @staticmethod
-    def pl_to_stim(op):
+    def _pl_to_stim(op):
```

Comment:
If it's a private static method, does it really need to live on the class? Or could it just be a private helper function at the top of the file?